velox/expression/tests/ArgumentTypeFuzzerTest.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/expression/tests/ArgumentTypeFuzzer.h"
+
+#include <gtest/gtest.h>
+
+#include "velox/expression/SignatureBinder.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::test {
+
+void testBindingSuccess(
+    const std::shared_ptr<exec::FunctionSignature>& signature,
+    const TypePtr& returnType,
+    const std::vector<TypePtr>& expectedArgumentTypes) {
+  std::mt19937 seed{0};
+  ArgumentTypeFuzzer fuzzer{*signature, seed};
+
+  ASSERT_TRUE(fuzzer.fuzzArgumentTypes(returnType));
+
+  auto& argumentTypes = fuzzer.argumentTypes();
+  ASSERT_EQ(expectedArgumentTypes.size(), argumentTypes.size());
+
+  auto& argumentSignatures = signature->argumentTypes();
+  for (int i = 0; i < expectedArgumentTypes.size(); ++i) {
+    ASSERT_TRUE(expectedArgumentTypes[i]->equivalent(*argumentTypes[i]));
+  }
+}
+
+void testBindingFailure(
+    const std::shared_ptr<exec::FunctionSignature>& signature,
+    const TypePtr& returnType) {
+  std::mt19937 seed{0};
+  ArgumentTypeFuzzer fuzzer{*signature, seed};
+
+  ASSERT_FALSE(fuzzer.fuzzArgumentTypes(returnType));
+}
+
+TEST(SignatureTemplateBinderTest, concreteSignature) {
```

Comment:
- Test name needs updating. 
- Consider adding test class and move testBindingFailure/Success into it. 
- It would be nice to add tests for ReverseSignatureBinder.